ast.c

```diff
@@ -287,6 +292,8 @@ static const ast_flag_info flag_info[] = {
 	{ ZEND_AST_CONDITIONAL, 1, conditional_flags },
 };
 
+// NOTE(tandre) in php 8, to get a writeable pointer to a property, Z_OBJ_HT_P(zv)->get_property_ptr_ptr(Z_OBJ_P(zv), AST_STR(str_propname), BP_VAR_RW, AST_CACHE_SLOT_PROPNAME); can be used.
```

Comment:
Just using `OBJ_PROP_NUM` and relying on specific offsets would be more efficient.

Reply:
Okay, wasn't familiar with how to do that, will look when I have time